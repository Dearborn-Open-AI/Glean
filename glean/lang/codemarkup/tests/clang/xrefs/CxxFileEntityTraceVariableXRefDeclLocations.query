query: codemarkup.cxx.CxxFileEntityTraceVariableXRefDeclLocations _
transform: [gensort, []]
