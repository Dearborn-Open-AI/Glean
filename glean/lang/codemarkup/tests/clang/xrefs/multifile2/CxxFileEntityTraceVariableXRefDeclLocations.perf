{ "@generated": null, "cxx1.DeclarationNameSpan.5": 1, "cxx1.FileXRefs.5": 2 }