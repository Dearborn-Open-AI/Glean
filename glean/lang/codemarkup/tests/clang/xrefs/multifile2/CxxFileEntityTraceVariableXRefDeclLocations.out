[
  "@generated",
  {
    "key": {
      "trace": {
        "key": {
          "file": { "key": "b.h" },
          "fixed": [
            {
              "target": {
                "declaration": {
                  "record_": {
                    "key": {
                      "name": {
                        "key": {
                          "name": { "key": "R" },
                          "scope": { "global_": { } }
                        }
                      },
                      "kind": { "struct_": { } },
                      "source": {
                        "file": { "key": "b.h" },
                        "lineBegin": 9,
                        "columnBegin": 1,
                        "lineEnd": 9,
                        "columnEnd": 11
                      }
                    }
                  }
                }
              },
              "ranges": [ { "offset": 248, "length": 1 } ]
            }
          ],
          "variable": [ [ { "offset": 255, "length": 1 } ] ]
        }
      },
      "source": {
        "function_": {
          "key": {
            "name": {
              "key": {
                "name": { "key": { "name": { "key": "g" } } },
                "scope": { "global_": { } }
              }
            },
            "signature": {
              "key": { "returns": { "key": "void" }, "parameters": [ ] }
            },
            "source": {
              "file": { "key": "a.h" },
              "lineBegin": 12,
              "columnBegin": 1,
              "lineEnd": 15,
              "columnEnd": 1
            }
          }
        }
      },
      "location": {
        "name": "g",
        "file": { "key": "a.h" },
        "location": {
          "range": {
            "file": { "key": "a.h" },
            "lineBegin": 12,
            "columnBegin": 1,
            "lineEnd": 15,
            "columnEnd": 1
          }
        },
        "span": { "start": 253, "length": 1 }
      }
    }
  }
]